usr/sbin/pkcsslotd/mutex.c

```diff
@@ -27,6 +27,9 @@ int CreateXProcLock(void)
     struct group *grp;
     mode_t mode = (S_IRUSR | S_IRGRP);
 
+    if (xplfd == -1)
+        xplfd = open(OCK_API_LOCK_FILE, O_RDONLY);
```

Comment:
is the mode parm intentionally left out here?

Reply:
Yes, the mode parameter only has a meaning if O_CREAT (or O_TMPFILE ) is specified (i.e. the file is to be created), otherwise it is ignored. So no need to specify it here, since it is using O_RDONLY to try to open an existing file.